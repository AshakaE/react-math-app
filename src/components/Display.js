import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ result }) => (
  <div className="flex w-[700px] justify-end bg-gray-600 h-[100px] text-white font-bold pt-8 pr-5 text-5xl">{result}</div>
);

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};
export default Display;
