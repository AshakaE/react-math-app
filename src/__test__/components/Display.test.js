import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Display from '../../components/Display';
import '@testing-library/jest-dom';

describe('Display', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Display />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with default value 0', () => {
    const { container } = render(<Display />);
    const display = container.querySelector('div');
    expect(display).toHaveTextContent('0');
  });
  it('renders the corerct result given as prop', () => {
    const { container } = render(<Display result="58" />);
    const display = container.querySelector('div');
    expect(display).toHaveTextContent('58');
  });
});
