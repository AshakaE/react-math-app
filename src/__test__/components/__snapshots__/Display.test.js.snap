// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div
  className="flex w-[700px] justify-end bg-gray-600 h-[100px] text-white font-bold pt-8 pr-5 text-5xl"
>
  0
</div>
`;
